backend/gateway/meta/meta.go

```diff
@@ -171,3 +173,38 @@ func resolvePattern(message descriptor.Message, pattern *apiv1.Pattern) *auditv1
 
 	return &auditv1.Resource{TypeUrl: pattern.TypeUrl, Id: resourceName}
 }
+
+// isValidInterface will return true if the type is proto.message and value is not nil
+func isValidInterface(body interface{}) bool {
+	switch v := body.(type) {
+	// type we want is proto.message
+	case proto.Message:
+		// want to use a value that is not nil
+		if !reflect.ValueOf(v).IsNil() {
```

Comment:
not sure what case we'd hit this in, but have been wondering if we also want to check if the proto message is empty.

if it's okay to use the v2 message type (saw an import of the package [here](https://github.com/lyft/clutch/blob/main/backend/gateway/log/log.go#L8)), we can use the ProtoReflect().[IsValid](https://godoc.org/google.golang.org/protobuf/reflect/protoreflect#Cardinality.IsValid)() method, which checks if it is an empty object (based on whether the proto message has defined required fields ) and for a nil value

Reply:
update: RE my first comment, we run into this scenario with https://github.com/lyft/clutch/blob/main/api/k8s/v1/k8s.proto#L232 which is an empty message. this is what is shows up as in the logger event:

`new audit event	{"serviceName": "clutch.service.audit.sink.logger", "event": {"occurredAt":"2020-10-22T16:46:21.423558Z","event":{"username":"UNKNOWN","serviceName":"clutch.k8s.v1.K8sAPI","methodName":"ResizeHPA","type":"UPDATE","status":{},"resources":[{"typeUrl":"clutch.k8s.v1.HPA","id":"kind-clutch/envoy-staging/envoy-main"}],"requestMetadata":{"body":{"@type":"type.googleapis.com/clutch.k8s.v1.ResizeHPARequest","clientset":"kind-clutch","cluster":"kind-clutch","namespace":"envoy-staging","name":"envoy-main","sizing":{"min":7,"max":7}}},"responseMetadata":{"body":{"@type":"type.googleapis.com/clutch.k8s.v1.ResizeHPAResponse"}}}}}` 

empty object != nil